src/ktopt_planning_context.cpp

```diff
@@ -48,6 +48,62 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
   const auto joint_model_group = getPlanningScene()->getRobotModel()->getJointModelGroup(getGroupName());
   RCLCPP_INFO_STREAM(getLogger(), "Planning for group: " << getGroupName());
 
+  // extract position and velocity bounds
+  std::vector<double> lower_position_bounds;
```

Comment:
Try initialize all of these with the known size (number of dofs) for efficiency.

If you initialize them, change the `push_back`s for modifying existing elements.

Or you can `reserve()` them in which case you do keep pushing back.